src/frontend/qt_sdl/main.cpp

```diff
@@ -1573,28 +1611,31 @@ void MainWindow::onOpenFileArchive()
 {
     emuThread->emuPause();
 
-    QString filename = QFileDialog::getOpenFileName(this,
+    QString archiveFileName = QFileDialog::getOpenFileName(this,
                                                     "Open ROM Archive",
                                                     Config::LastROMFolder,
                                                     "Archived ROMs (*.zip *.7z *.rar *.tar *.tar.gz *.tar.xz *.tar.bz2);;Any file (*.*)");
-    if (filename.isEmpty())
+    if (archiveFileName.isEmpty())
     {
         emuThread->emuUnpause();
         return;
     }
 
     printf("Finding list of ROMs...\n");
-    QVector<QString> archiveROMList = Archive::ListArchive(filename.toUtf8().constData());
+    QVector<QString> archiveROMList = Archive::ListArchive(archiveFileName.toUtf8().constData());
+    QByteArray *romBuffer = new QByteArray();
+    QString romFileName; // file name inside archive
+
     if (archiveROMList.size() > 2)
     {
         archiveROMList.removeFirst();
         QString toLoad = QInputDialog::getItem(this, "melonDS",
                                   "The archive was found to have multiple files. Select which ROM you want to load.", archiveROMList.toList(), 0, false);
         printf("Extracting '%s'\n", toLoad.toUtf8().constData());
-        QVector<QString> extractResult = Archive::ExtractFileFromArchive(filename.toUtf8().constData(), toLoad.toUtf8().constData());
+        QVector<QString> extractResult = Archive::ExtractFileFromArchive(archiveFileName.toUtf8().constData(), toLoad.toUtf8().constData(), romBuffer);
```

Comment:
I don't know why I converted the QString results to const char*. It is only converted back to QString anyway.
```suggestion
        QVector<QString> extractResult = Archive::ExtractFileFromArchive(archiveFileName, toLoad, romBuffer);
```